addons/binding/org.openhab.binding.nibeheatpump/src/main/java/org/openhab/binding/nibeheatpump/internal/message/NibeHeatPumpBaseMessage.java

```diff
@@ -0,0 +1,97 @@
+/**
+ * Copyright (c) 2014-2017 by the respective copyright holders.
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.openhab.binding.nibeheatpump.internal.message;
+
+import javax.xml.bind.DatatypeConverter;
+
+import org.openhab.binding.nibeheatpump.internal.NibeHeatPumpException;
+import org.openhab.binding.nibeheatpump.internal.protocol.NibeHeatPumpProtocol;
+
+/**
+ * The {@link NibeHeatPumpBaseMessage} define abstract class for Nibe messages. All message implementations should
+ * extend this class.
+ *
+ *
+ * @author Pauli Anttila - Initial contribution
+ */
+public abstract class NibeHeatPumpBaseMessage implements NibeHeatPumpMessage {
+
+    public static MessageType getMessageType(byte messageType) {
+        for (MessageType p : MessageType.values()) {
+            if (p.toByte() == messageType) {
+                return p;
+            }
+        }
+        return MessageType.UNKNOWN;
```

Comment:
Binding will receive lot of message which are not interested by the binding (at to the moment at least). Exception creation can be expensive.